src/index.js

```diff
@@ -0,0 +1,159 @@
+const state = {
+    defaultCity: 'San Francisco',
+    city: 'San Francisco, CA',
+    temperature: 68,
+    lat: 37.7790262,
+    lon: -122.419906,
+};
+
+const skyState = {
+    sunny: '🌞🌞🌞🌞🌞🌞🌞🌞🌞',
+    cloudy: '☁️☁️ ☁️ ☁️☁️ ☁️ 🌤 ☁️☁️',
+    rainy: '🌧🌈🌧🌧💧🌧🌦🌧💧🌧🌧',
+    snowy: '🌨❄️🌨🌨❄️❄️🌨❄️🌨❄️❄️🌨🌨'
+};
+
+const landState = {
+    '80+': '🌵🌵🌵🌵🌵🌵🌵🌵🌵',
+    '70+': '🌴🌴🌴🌴🌴🌴🌴🌴🌴',
+    '60+': '🌳🌳🌳🌳🌳🌳🌳🌳🌳',
+    '59-': '🏔️☃️🏔️🏔️🥶🏔️🏔️🏔️☃️'
+};
+
+const tempState = {
+    '80+': 'red',
+    '70+': 'orange',
+    '60+': 'gold',
+    '50+': 'green',
+    '49-': 'teal'
+};
+
+const locationUrl = 'http://127.0.0.1:5000/location';
+const weatherUrl = 'http://127.0.0.1:5000/weather';
+
+const increaseTemp = () => {
+    const tempContainer = document.querySelector('#tempValue');
+    state.temperature++;
+    tempContainer.textContent = state.temperature;
+    changeTempColor();
+    changeLandscape();
+};
+
+const decreaseTemp = () => {
+    const tempContainer = document.querySelector('#tempValue');
+    state.temperature--;
+    tempContainer.textContent = state.temperature;
+    changeTempColor();
+    changeLandscape();
+};
+
+const renameCityHeader = () => {
+    const headerCity = document.querySelector('#headerCityName');
+    const newCity = document.querySelector('#cityNameInput');
+    state.city = newCity.value;
+    headerCity.innerText = state.city;
+};
+
+const resetCity = () => {
+    const headerCity = document.querySelector('#headerCityName');
+    headerCity.innerText = state.defaultCity;
+    const newCity = document.querySelector('#cityNameInput');
+    newCity.value = state.defaultCity;
+};
+
+const getLatAndLon = () => {
+    axios.get(locationUrl, {
+        params: {
+            q: state.city,
+            format: "json",
+        }
+    })
+        .then((res) => {
+            state.lat = res.data[0].lat;
+            state.lon = res.data[0].lon;
+        })
+        .then(getCurrentWeather())
+        .catch((err) => {
+            console.log("Failed to load latitude and longitude")
+        })
+};
+
+const tempConvertKtoF = (temp) => (temp - 273.15) * (9 / 5) + 32;
+
+const setTempToRealTimeTemp = () => {
+    const tempContainer = document.querySelector('#tempValue');
+    tempContainer.innerText = state.temperature;
+    changeTempColor();
+    changeLandscape();
+};
+
+const getCurrentWeather = () => {
```

Comment:
Assuming we had `getLatAndLon` that takes a city name and returns a promise to lat/lon coordinates, and a function `getCurrentWeather` that takes lat/lon coordinates and returns a promise to the current temperature, we could make a function that combines the two, `getTemperatureForCity`
```js
const getTemperatureForCity = (cityName) => {
    return getLatAndLon(cityName)
        .then({ lat, lon } => {
            return getCurrentWeather(lat, lon)
        })
```
This could be called in an event handler to get the temperate and update the UI as follows
```js
    return getTemperatureForCity("Seattle")
        .then(tempF => {
            state.temperature = tempF;
            setTempToRealTimeTemp()
        });
```